src/Microsoft.DotNet.Build.Tasks/PackageFiles/VersionTools.targets

```diff
@@ -64,7 +64,7 @@
     <ItemGroup>
       <_UpgradableSubmodule Include="@(SubmoduleConfiguration)"
                             Condition="'%(SubmoduleConfiguration.Branch)' != '' AND
-                                       '%(SubmoduleConfiguration.SourceBuildAutoUpdate)' == 'true'" />
+                                       '%(SubmoduleConfiguration.VersionToolsAutoUpdate)' != 'false'" />
```

Comment:
I think you need to change the places in ReadGitConfigFile as well.

Reply:
Indeed, fixed up and tested in source-build.